### ISSUE TITLE
Notifications not shown when having Actions in Outlook Desktop, when triggered by events

### ISSUE THREAD

**Issue by bogdanst24** [OfficeDev/office-js#4141]
# Provide required information needed to triage your issue

## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: Windows
* Host [Excel, Word, PowerPoint, etc.]: Outlook
* Office version number: Version 2402 Build 16.0.17328.20068

## Expected behavior

Notification with action is displayed in Outlook Desktop when asked. 

## Current behavior

In Outlook Desktop (not the new one), actionable notifications are not being shown anymore when run from event-based activation (e.g. new messages). The `addAsync` or `replaceAsync` actions throw an exception.

# Provide additional details

The following snippet, when run from an event-based add-in, throws
```
[TypeError: Unable to get property 'hostPlatform' of undefined or null reference]	
```

```
    const details = {
      message: 'test message',
      type: Office.MailboxEnums.ItemNotificationMessageType.InsightMessage,
      icon: 'icon_resource',
      actions: [
        {
          actionText: 'test action',
          actionType: Office.MailboxEnums.ActionType.ShowTaskPane,
          commandId: 'msgReadOpenPaneButton',
          contextData: undefined,
        },
      ],
    };
    
    mailbox.item.notificationMessages.addAsync(key, details,asyncResultCallback(resolve, reject));
```
Tested also with the example provided in the [documentation](https://learn.microsoft.com/en-us/javascript/api/outlook/office.notificationmessageaction?view=outlook-js-preview#examples), and nothing changes

There also seems to be a problem with the `Icon` validation. Adding an `InformationMessage` without an icon returns a required icon exception, even though the icon is not required in this context based on [documentation](https://learn.microsoft.com/en-us/javascript/api/outlook/office.notificationmessagedetails?view=outlook-js-preview#outlook-office-notificationmessagedetails-icon-member).
```
{ [Sys.ArgumentNullException: Sys.ArgumentNullException: Value cannot be null.
Parameter name: icon] name: 'Sys.ArgumentNullException', paramName: 'icon' }
```

**Comment by bogdanst24** [comment]
@exextoc some help here?

**Comment by Oleg-O** [comment]
Hi @bogdanst24, apologies for missing to follow-up here for so long, we're working to get better with our responsiveness.

On the InsightMessage issue, I believe this was fixed in build 16.0.17328.20128, can you try this or later build and let me know whether this works now? I've tried this with latest builds on both monthly and semi-annual channels and it appears to be working well.

On the icon for InformationMessage, thank you for pointing it out. I believe this is kind of a minor issue but nonetheless I've logged an item on our end to track this (office:9477228) - we will either fix the code to make it optional as the documantation states or fix the docs to reflect current behavior.

**Comment by bogdanst24** [comment]
I can confirm it was fixed
